lib/shopify_cli/theme/extension/syncer.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+require_relative "syncer/extension_serve_job"
+
+module ShopifyCLI
+  module Theme
+    module Extension
+      class Syncer
+        attr_accessor :pending_updates, :latest_sync
+
+        def initialize(ctx, extension:)
+          @ctx = ctx
+          @extension = extension
+
+          @pool = ThreadPool.new(pool_size: 1)
+          @pending_updates = []
```

Comment:
I believe that even with the `pool_size` of `1`, we might face issues between the single thread and the main thread.

So, I believe we might use a `Queue` here, or a `Mutex` to coordinate changes in the `@pending_updates` (I'd follow the `Queue` approach because of the following paragraph).

Also, I believe we might keep the `@pending_updates` exposed with the `attr_reader` only (so the API user might rely on the `Queue#clear` to proper clear the object.